Marlin/src/module/planner.cpp

```diff
@@ -2005,7 +2005,7 @@ bool Planner::_populate_block(
   #if HAS_EXTRUDERS
     dm.e = (dist.e > 0);
     const float esteps_float = dist.e * e_factor[extruder];
-    const uint32_t esteps = ABS(esteps_float) + 0.5f;
+    const uint32_t esteps = ABS(esteps_float);
```

Comment:
This rounding to the nearest number of steps has been in the code since prehistoric times. Are you sure it should be removed?

Reply:
I didn't understand why we truncate all of the other axis steps but round the esteps so I removed it. I think you're the one who added it 7 years ago, you'd know better than anyone. I kinda forgot I made this change -- it wasn't really supposed to be part of this PR.

Reply:
I think this changed should be reverted.

Reply:
Okay. Why do we currently round esteps and not other axis?

Reply:
I'll keep looking into that history. Meanwhile, I'm not sure. It was likely a considered choice at some point. Maybe it is to to mitigate the use of `e_factor` and the introduction of further `float` imprecision. Maybe to have the nearest magnitude correspondence between `esteps_float` and `esteps`. It's hard to say whether it's of any import without looking at some different situations. I tend to agree that a single half-E-step probably doesn't matter in the grand scale of E-steps.